Gems/PhysX/Code/Source/HeightfieldColliderComponent.h

```diff
@@ -55,52 +54,28 @@ namespace PhysX
         void Activate() override;
         void Deactivate() override;
 
-        void SetShapeConfiguration(const AzPhysics::ShapeColliderPair& shapeConfig);
+        void SetColliderConfiguration(const Physics::ColliderConfiguration& colliderConfig);
 
     protected:
         // ColliderComponentRequestBus
         AzPhysics::ShapeColliderPairList GetShapeConfigurations() override;
         AZStd::vector<AZStd::shared_ptr<Physics::Shape>> GetShapes() override;
 
-        // ColliderShapeRequestBus
-        AZ::Aabb GetColliderShapeAabb() override;
-        bool IsTrigger() override
-        {
-            // PhysX Heightfields don't support triggers.
-            return false;
-        }
-
         // CollisionFilteringRequestBus
         void SetCollisionLayer(const AZStd::string& layerName, AZ::Crc32 filterTag) override;
         AZStd::string GetCollisionLayerName() override;
         void SetCollisionGroup(const AZStd::string& groupName, AZ::Crc32 filterTag) override;
         AZStd::string GetCollisionGroupName() override;
         void ToggleCollisionLayer(const AZStd::string& layerName, AZ::Crc32 filterTag, bool enabled) override;
 
-        // SimulatedBodyComponentRequestsBus
-        void EnablePhysics() override;
-        void DisablePhysics() override;
-        bool IsPhysicsEnabled() const override;
-        AZ::Aabb GetAabb() const override;
-        AzPhysics::SimulatedBodyHandle GetSimulatedBodyHandle() const override;
-        AzPhysics::SimulatedBody* GetSimulatedBody() override;
-        AzPhysics::SceneQueryHit RayCast(const AzPhysics::RayCastRequest& request) override;
-
-        // HeightfieldProviderNotificationBus
-        void OnHeightfieldDataChanged(const AZ::Aabb& dirtyRegion, 
-            Physics::HeightfieldProviderNotifications::HeightfieldChangeMask changeMask) override;
-
     private:
         AZStd::shared_ptr<Physics::Shape> GetHeightfieldShape();
 
-        void ClearHeightfield();
-        void InitHeightfieldShapeConfiguration();
-        void InitStaticRigidBody();
-        void RefreshHeightfield(Physics::HeightfieldProviderNotifications::HeightfieldChangeMask changeMask,
-            const AZ::Aabb& dirtyRegion = AZ::Aabb::CreateNull());
-
-        AzPhysics::ShapeColliderPair m_shapeConfig;
-        AzPhysics::SimulatedBodyHandle m_staticRigidBodyHandle = AzPhysics::InvalidSimulatedBodyHandle;
-        AzPhysics::SceneHandle m_attachedSceneHandle = AzPhysics::InvalidSceneHandle;
+        //! Stores collision layers, whether the collider is a trigger, etc.
+        AZStd::shared_ptr<Physics::ColliderConfiguration> m_colliderConfig{ aznew Physics::ColliderConfiguration() };
+        //! Stores all of the cached information for the heightfield shape.
+        AZStd::shared_ptr<Physics::HeightfieldShapeConfiguration> m_shapeConfig{ aznew Physics::HeightfieldShapeConfiguration() };
```

Comment:
The shape config is currently just more of a "placeholder" container that gets filled in at runtime from the terrain system with data on how many rows and columns it has, and what the samples in it are. We don't _want_ that data serialized because the serialized data wouldn't necessarily contain any terrain data that's loaded dynamically or generated or modified at runtime. @SergeyAMZN is starting work on heightfield baking that _will_ allow for serialized heightfield data in the cases that a user knows the data won't change and doesn't need to be calculated at runtime by the heightfield system.

Reply:
Awesome! Make sense, thanks for the explanation.